module/VuFind/src/VuFind/Controller/Plugin/ResultScroller.php

```diff
@@ -63,32 +73,49 @@ class ResultScroller extends AbstractPlugin
      */
     protected $resultsManager;
 
+    /**
+     * Search memory
+     *
+     * @var SearchMemory
+     */
+    protected $searchMemory;
+
+    /**
+     * Currently active scroll data
+     *
+     * @var \StdClass
+     */
+    protected $data = null;
+
     /**
      * Constructor. Create a new search result scroller.
      *
      * @param SessionContainer $session Session container
      * @param ResultsManager   $rm      Results manager
      * @param bool             $enabled Is the scroller enabled?
+     * @param SearchMemory     $sm      Search memory
      */
     public function __construct(
         SessionContainer $session,
         ResultsManager $rm,
-        $enabled = true
+        $enabled = true,
+        SearchMemory $sm = null
     ) {
         $this->enabled = $enabled;
 
         // Set up session namespace for the class.
-        $this->data = $session;
+        $this->session = $session;
 
         $this->resultsManager = $rm;
+        $this->searchMemory = $sm;
     }
 
     /**
      * Initialize this result set scroller. This should only be called
      * prior to displaying the results of a new search.
      *
-     * @param \VuFind\Search\Base\Results $searchObject The search object that was
-     * used to execute the last search.
+     * @param Results $searchObject The search object that was used to execute the
+     * last search.
      *
      * @return bool
```

Comment:
What does this return value even mean? I wonder if we should explain something in the comments here. (I gather it's true when there are scroll results available, but I wouldn't typically expect an init() method to return a boolean).

Reply:
I can't see the return value being useful either, but the tests check it so I left it in for now. I added an explanation for it.